pkg/controllers/customroute/custom_route_conditions.go

```diff
@@ -43,27 +57,46 @@ func findCondition(conditions []metav1.Condition, conditionType string) *metav1.
 	return nil
 }
 
-func checkErrorsConfiguringCustomRoute(errors []error) []metav1.Condition {
-	if len(errors) != 0 {
-		now := metav1.Now()
-		return []metav1.Condition{
-			{
-				LastTransitionTime: now,
-				Type:               "Degraded",
-				Status:             metav1.ConditionTrue,
-				Reason:             "CustomRouteError",
-				Message:            fmt.Sprintf("Error Configuring custom route: %v", errors),
-			},
+func checkErrorsConfiguringCustomRoute(errs []error) []metav1.Condition {
+	now := metav1.Now()
+	var conditions []metav1.Condition
+	var degradedErrors, upgradeableErrors []error
+	for _, err := range errs {
+		conditions = []metav1.Condition{
```

Comment:
you're rewriting this in each loop? I assume you might have intended to do something else?

Reply:
that was a copy-paste mistake, thanks for catching.